Xamarin.PropertyEditing.Tests/MockObjectEditor.cs

```diff
@@ -177,6 +177,19 @@ public Task<IReadOnlyList<ITypeInfo>> GetAssignableTypesAsync (IPropertyInfo pro
 					softType.GetProperty ("Value").SetValue (softValue, v);
 					softType.GetProperty ("Source").SetValue (softValue, value.Source);
 				}
+
+				if (typeof(T).Name == "IReadOnlyList`1") {
+					var list = (IReadOnlyList<int>) value.Value;
```

Comment:
Looks fine to start with just a list of ints, but I'd still put some safeguards in there and at least throw not implemented if that cast fails.